scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java

```diff
@@ -1249,65 +1385,110 @@ public DtlsConnectorConfig build() {
 			if (config.staleConnectionThreshold == null) {
 				config.staleConnectionThreshold = DEFAULT_STALE_CONNECTION_TRESHOLD;
 			}
-			if (config.certificateVerifier == null && config.trustStore != null) {
-				config.certificateVerifier = new StaticCertificateVerifier(config.trustStore);
-			}
 			if (config.sniEnabled == null) {
 				config.sniEnabled = Boolean.TRUE;
 			}
 			if (config.verifyPeersOnResumptionThreshold == null) {
 				config.verifyPeersOnResumptionThreshold = DEFAULT_VERIFY_PEERS_ON_RESUMPTION_THRESHOLD_IN_PERCENT;
 			}
+			if (config.certificateVerifier == null && config.trustStore != null) {
+				config.certificateVerifier = new StaticCertificateVerifier(config.trustStore);
+			}
+			if (config.trustCertificateTypes == null) {
+				if (config.trustedRPKs != null) {
+					config.trustCertificateTypes = ArrayUtils.add(config.trustCertificateTypes,
+							CertificateType.RAW_PUBLIC_KEY);
+				}
+				if (config.certificateVerifier != null) {
+					config.trustCertificateTypes = ArrayUtils.add(config.trustCertificateTypes, CertificateType.X_509);
+				}
+			} else {
```

Comment:
For better readability, I think this could be better to clearly separated default value initialization and consistency check.

Reply:
OK